src/ClubMappingData.php

```diff
@@ -0,0 +1,120 @@
+<?php declare(strict_types=1);
+
+namespace Football;
+
+class ClubMappingData
+{
+    public const DEFAULT_JSON_FILE = 'club_name_mapping.json';
+
+    /**
+     * Club name mapping
+     * @var string[]
+     */
+    private $clubNameMapping;
+
+    /**
+     * constructor
+     * @param array|null $data
+     */
+    public function __construct(?array $data = null)
+    {
+        if ($data !== null && ! empty($data)) {
```

Comment:
!empty is enough. https://stackoverflow.com/questions/4559925/why-check-both-isset-and-empty/4560099